Project Information Project Title | Student Name(s) | Enrollment Number(s) | Group Number(s)
Project - 01
Project Title: Data Exploration with Python using Pandas & Numpy Libraries
Submitted By: Saumya Raghuvanshi(055040), Jhalki Kulshrestha(055017)
Group No: 19
Program: PGDM-BDA-05
Faculty: Prof Amarnath Mitra
Subject: Data Exploration & Visualization using Python (DEVP)

Description of Data

-> Data Source & Size
      .Data Source: The dataset used in this project provides detailed information about
      international trade transactions, with a focus on imports and exports.This dataset
      allows the user to explore various statistics such as mean, median, mode, maximum,
      and minimum values.The data likely comes from a business or organization involved in
      international trade either from a company's internal database or a global trade
      platform that tracks import and export transactions.
      .Size: 15,000 observations (rows) and 16 variables (columns). Each record represents 
      a single import/export transaction, and the data encompasses various transactions
      over months/years.

-> Data Type: The dataset is cross-sectional. It captures data from a single point in time for each transaction, providing a snapshot rather than tracking changes over time.

-> Data Dimension:
      .Number of Variables: 16
      .Number of Observations: 15,000

-> Data Variable Type:
      .Text: Variables such as Transaction_ID, Country, Product, Import_Export, Date, Category, Port,
      Shipping_Method, Supplier, Customer, and Payment_Terms contain text data.
      .Number:
       Integer: Quantity, Customs_Code, and Invoice_Number
       Decimal: Value (transaction value) and Weight (weight of the shipment)
       
-> Data Variable Category:
      .Index: Transaction_ID (unique identifier for each transaction), Invoice_Number
      (unique invoice identification)
      .Categorical Nominal Variable: Variables such as Country, Product, Import_Export,
      Category, Port,
      Shipping_Method, Supplier, Customer, and Payment_Terms that do not have an inherent
      order.
         Import/Export Status (Categorical Variable): This column distinguishes whether a
         transaction involves the importing or exporting of goods. This classification is
         crucial for the analysis as it allows the separation of inbound and outbound trade
         flows. It enables a comparison between the characteristics of imports versus
         exports, such as differences in transaction values or quantities. By analyzing the
         distribution of import and export transactions, businesses can identify trade
         imbalances (i.e., whether they import significantly more than they export or vice
         versa).
         Shipping Method (Categorical Variable): This column indicates the mode of
         transportation used to ship goods, such as air, sea, or land. The shipping method
         impacts delivery time, cost, and reliability. Different goods require different
         shipping methods based on factors like perishability, cost efficiency,
         and distance. Understanding the distribution of shipping methods used can help
         companies optimize their logistics by choosing cost-effective or faster shipping
         routes. Analyzing how often each method is used can also inform decisions on
         contract negotiations with logistics providers.
         Payment Terms (Categorical Variable): This column outlines the terms of payment
         for each transaction, such as pre-paid, post-paid, or credit terms.This Significes
         that the payment terms directly affect cash flow and financial planning. Longer
         credit periods may affect liquidity, while upfront payments reduce
         financial risks.By examining the payment terms used for different transactions,
         businesses can optimize their financial policies. For instance, frequent use of
         credit terms may warrant tighter financial management, while prepaid terms offer
         more immediate cash flow benefits.
      .Non-Categorical: Variables like Quantity, Value, Weight, and Customs_Code are
      numerical and continuous in nature, representing measurable quantities.

Project Objectives | Problem Statements

The project involves analyzing import and export transaction data to uncover patterns, relationships, and key metrics such as transaction values, quantities, and correlations between variables like shipping methods and payment terms. It focuses on exploring trade data to extract meaningful insights, particularly by examining transaction patterns, the quantities of goods traded, and the potential connections between various business practices.

The primary objective is to conduct statistical analysis to better understand key trade metrics and relationships among factors influencing business operations. This includes:
- Calculating key statistics (mean, median, mode) to summarize typical transaction values.
- Identifying extreme values (maximum and minimum) for quantities traded.
- Performing hypothesis testing (e.g., Chi-squared tests) to explore relationships between factors such as shipping methods and payment terms, offering deeper operational insights.

Python libraries are used extensively for data analysis and hypothesis testing:
- Pandas: For data manipulation and analysis.
- NumPy: To compute summary statistics such as mean and median.
- SciPy/Statsmodels: For advanced statistical testing, like the Chi-squared test, to assess relationships between categorical variables.

The findings of this project are particularly valuable for organizations involved in global supply chains and trade logistics. Through trade data analysis, businesses can:
- Make informed decisions on pricing, logistics, and payment strategies.
- Identify risk areas, such as significant fluctuations in export values or inconsistent shipping methods.
- Adjust operational and financial strategies to maximize profits, mitigate risks, and ensure smooth cross-border operations.

In summary, the project emphasizes the critical role of data and statistical analysis in understanding trade dynamics and highlights the value of leveraging analytics to manage international business operations effectively.

Observations | Findings

1. Dataset Composition and Structure:
The dataset contains 15,000 rows and 16 columns, representing import and export transactions, with a sample of 2,001 records used for the analysis.
Key Variables:
Categorical: Country, Product, Import_Export, Shipping_Method, Payment_Terms, Supplier, etc.
Numerical: Quantity, Value, Weight, Customs_Code.
Observations/Findings:
The dataset offers a wide range of transaction-level information, including specific countries, shipping methods, and payment terms, providing a detailed view of global trade operations.
The dataset allows for segmentation by import and export transactions, which is crucial for understanding the dynamics of trade from both perspectives.

Recommendations:
Data Enrichment: Consider enriching the dataset with additional variables like exchange rates, transportation costs, and tariff details. This can further improve the quality of insights, especially when analyzing trade profitability and shipping logistics.

2. Import vs. Export Transactions:
The analysis shows distinct differences between import and export transactions.
Exports generally had higher transaction values, suggesting the country’s export items might be of higher value compared to imports.
Observations/Findings:
Exports tend to involve high-value goods, possibly electronics, machinery, or other manufactured products.
Imports, while frequent, generally involve goods with a lower average value than exports.

Recommendations:
Export Strengthening: Focus on boosting export capabilities, especially in high-value sectors like electronics and machinery, which appear to be driving significant export revenues.
Import Strategy: For imports, investigate potential cost-saving strategies, such as negotiating better deals with suppliers or exploring alternative sourcing options to reduce costs.

3. Shipping Methods:
Sea Freight is the most commonly used shipping method, followed by Air Freight. The preference for shipping method varies significantly depending on the type of goods being traded and the transaction value.
Observations/Findings:
Sea Freight dominates, likely due to its cost-effectiveness for large shipments of goods with lower value-to-weight ratios (e.g., raw materials, bulk products).
Air Freight is primarily used for goods with higher value but lower volume, where the speed of delivery outweighs cost considerations.

Recommendations:
Optimize Sea Freight: Since sea freight is already the most used method, businesses should explore further optimizations, such as consolidating shipments to reduce costs or negotiating long-term contracts with shipping providers.
Selective Air Freight Use: While air freight is more expensive, it should be strategically used for high-margin or time-sensitive products. Businesses should aim to optimize air freight by choosing faster customs clearance or expedited services for urgent shipments.

4. Country-Specific Trends:
The data highlights that a few countries, notably China and the USA, account for the largest number of transactions. These countries dominate in both imports and exports.
Observations/Findings:
China is the top partner for both imports and exports, reflecting its central role in global trade.
Some countries are underrepresented in the dataset, suggesting potential untapped markets.

Recommendations:
Diversify Trade Partners: While China and the USA are crucial trade partners, businesses should consider expanding into new markets to reduce dependency on a few key players. Emerging markets, particularly in Africa and Southeast Asia, may offer growth opportunities.
Trade Agreements: Strengthen trade agreements with top partners (e.g., China and the USA) to maintain competitiveness, especially in areas like tariff negotiations and customs processes.

5. Payment Terms:
The analysis explores different payment terms, with a focus on how they relate to transaction value. Common terms include Net 30 and Letters of Credit.
Observations/Findings:
Higher-value transactions are often associated with more secure payment terms (e.g., Letters of Credit), while lower-value transactions may use Net 30 or other flexible terms.
Delays in payments or inconsistent payment terms can introduce financial risks to the business, especially when trading internationally.

Recommendations:
Tailored Payment Terms: Encourage the use of secure payment terms (e.g., Letters of Credit) for high-value transactions to mitigate financial risks. For smaller transactions, offering more flexible payment options may help attract a broader customer base.
Risk Mitigation: Introduce a risk-based approach to managing payment terms, ensuring that higher-risk markets or transactions are backed by stricter payment conditions.

6. Transaction Value and Quantity Analysis:
Detailed descriptive statistics were computed for transaction value, quantity, and weight.
The analysis found significant variability in the value of transactions, with some high-value outliers in the dataset.
Observations/Findings:
There is a wide range of transaction values, with some exports showing extremely high values, likely reflecting luxury goods, specialized machinery, or high-tech exports.
Quantities and weights also vary widely, but bulk shipments (higher quantities) tend to involve lower-value goods (e.g., raw materials).

Recommendations:
Focus on High-Value Exports: Prioritize high-value goods for export, such as technology or machinery, which offer better profitability and potentially higher margins.
Inventory Management: For bulk goods, businesses should adopt efficient inventory and warehousing strategies to handle large quantities at lower costs. Implementing just-in-time inventory for high-volume, low-value goods can reduce storage costs and improve cash flow.

7. Transaction Patterns by Product:
Products traded across borders show distinct patterns. Some categories (e.g., electronics, machinery) dominate in exports, while other goods (e.g., raw materials) are frequently imported.
Observations/Findings:
Electronics and machinery are key export products, contributing significantly to the overall trade value.
Raw materials and lower-end goods tend to dominate imports, reflecting the country's reliance on external sources for basic inputs.

Recommendations:
Leverage Core Competencies in Exports: Businesses should continue to invest in the production and export of high-demand goods like electronics and machinery, while exploring opportunities to innovate or add value to their offerings.
Import Substitution: For imported goods that are core to manufacturing, explore domestic production or alternative suppliers to reduce reliance on external sources.

8. Risk and Opportunity Identification:
The analysis identifies areas of risk, such as high-value exports being concentrated in a few product categories or over-reliance on a small number of countries for trade.
Observations/Findings:
The concentration of high-value transactions in specific products or countries increases the risk of disruptions (e.g., political instability, trade wars).
Potential areas for expansion include underrepresented regions and product categories where competition may be less intense.

Recommendations:
Risk Mitigation: Develop risk management strategies to address the potential vulnerability of over-reliance on specific countries or products. Diversify both product lines and geographical markets.
Explore Growth Opportunities: Consider expanding into new product categories or regions to capitalize on less saturated markets. Use predictive analytics to identify emerging trends in global trade.

9. Correlation Between Variables:
The dataset allows for the exploration of potential correlations between variables like shipping methods, payment terms, transaction value, and product type.
Observations/Findings:
There may be significant correlations between shipping method and transaction value, as higher-value goods are more often shipped via air freight, while bulkier, lower-value goods use sea freight.
Payment terms and transaction size also appear to be related, with more secure terms used for higher-value transactions.

Recommendations:
Segment Trade Operations: 
Develop strategies that align shipping methods and payment terms with transaction size. For example, offer favorable air freight rates for premium customers or encourage the use of secure payment terms for larger transactions to minimize financial risk.
Custom Solutions:
Implement customized shipping and payment options based on transaction characteristics. Businesses should tailor their offerings to meet the needs of high-value, low-volume customers differently from those dealing with bulk shipments.

Overall Summary and Recommendations:

This detailed analysis of the import-export dataset reveals key trends, risks, and opportunities in global trade operations. The recommendations suggest that businesses should focus on optimizing their shipping logistics, diversifying trade markets, and managing financial risks through tailored payment terms and strategies. By focusing on high-value exports and exploring new markets, companies can enhance profitability while reducing exposure to external risks. Data-driven decision-making and predictive analytics will further allow businesses to stay ahead of global trade trends and adapt to the evolving market landscape.

Analysis of data

-> The mean, median, and mode of transaction values across all transactions calculated above provides valuable insights into the distribution of transaction values. Here’s what we can infer from the provided values:
1. Mean (5004.90):
The mean is the average of all transaction values. A mean of approximately 5004.90 suggests that the average transaction value is relatively high. However, the mean can be influenced by outliers or skewed data, so it’s essential to consider it alongside the median and mode.
2. Median (4989.12):
The median is the middle value when all transaction values are ordered. A median of 4989.12 indicates that half of the transactions are below this value and half are above. Since the median is very close to the mean, this suggests a relatively symmetrical distribution of transaction values, but it does not rule out the presence of outliers.
3. Mode (3673.92):
The mode is the most frequently occurring transaction value in the dataset. A mode of 3673.92 indicates that this value appears more often than any other. The mode being lower than both the mean and the median suggests that there may be a significant number of lower-value transactions, which could be influencing the average upward.

Overall Inferences:
Distribution Shape: The fact that the mean is higher than the median suggests a slight right skew in the distribution (more higher-value transactions). However, the closeness of the mean and median implies that the skew might not be extreme.
Presence of Outliers: The relatively high mean compared to the mode suggests that there are higher transaction values (possibly outliers) that are affecting the average. This could warrant further investigation to understand these high-value transactions better.
Transaction Value Trends: The mode indicates that a significant number of transactions are clustered around the value of 3673.92, which could be relevant for pricing strategies or market analysis.


-> From the mean and median transaction values calculated by product category, we can draw several inferences:
1. Clothing:
Mean (4873.99) vs. Median (4976.50):
The mean is slightly lower than the median, suggesting a slight leftward (negative) skew in the distribution. This may indicate that there are some lower-value transactions that are pulling the average down.
2. Electronics:
Mean (5221.99) vs. Median (5060.60):
Here, the mean is higher than the median, indicating a slight rightward (positive) skew. This suggests that there are higher-value transactions in electronics that are driving the average up.
3. Furniture:
Mean (5111.78) vs. Median (5167.39):
Similar to electronics, the mean is lower than the median. This suggests that there might be lower-value transactions affecting the average, leading to a slight leftward skew.
4. Machinery:
Mean (4896.36) vs. Median (4875.27):
The mean is slightly higher than the median, indicating a very mild rightward skew, but it’s close enough to suggest a relatively symmetrical distribution around the center.
5. Toys:
Mean (4941.23) vs. Median (4909.81):
The mean is again slightly higher than the median, suggesting a mild rightward skew similar to machinery.

Overall Inferences:
Skewness Insights:
Categories such as clothing and furniture show a slight leftward skew, indicating some low-value outliers. In contrast, electronics, machinery, and toys display a slight rightward skew, suggesting higher-value outliers may be present.

Central Tendency Comparison:
The mean and median values provide insights into the central tendency of each category. For example, electronics has the highest mean and median, indicating that this category tends to have the highest transaction values overall, followed by furniture and toys.

Data Distribution Considerations:
The differences between mean and median in each category suggest that when analyzing transaction values, it might be helpful to consider both measures of central tendency to understand the impact of skewed distributions and potential outliers.

-> From the range, standard deviation, and variance of the weights of shipped products calculated above, we can infer the following:
1. Range (4995.87):
A range of 4995.87 suggests a large spread between the lightest and heaviest products. This indicates a diverse set of product weights, which could have implications for shipping costs, handling, and storage requirements.
2. Standard Deviation (1462.02):
A standard deviation of 1462.02 indicates a high degree of variability in the weights of the products. This suggests that weights are widely dispersed around the mean, meaning some products are significantly lighter or heavier than the average.
3. Variance (2137514.84):
A variance of 2137514.84 confirms the high variability in product weights. 

Overall Inferences:
High Variability: The large range, along with the high standard deviation and variance, indicates that there is considerable variability in the weights of shipped products. This could mean that your product line includes both very light and very heavy items.
Impact on Shipping: The variability in weight might affect shipping strategies, costs, and logistics. For example, heavier products may require different handling procedures or shipping methods compared to lighter ones.
Potential for Optimization: If certain weights are significantly more common, it might be worth exploring whether shipping practices can be optimized for those weights, such as bulk shipping or specialized packaging.

Summary : 
Overall, the metrics suggest a diverse and variable distribution of product weights, which could have significant implications for logistics, costs, and operational efficiency. The high degree of variability indicates a need for careful planning and consideration in shipping and handling practices.

-> The skewness and kurtosis values calculated above provide insights into the distribution of export transaction values. Here’s what we can infer from each:
1. Skewness (0.0278):
A skewness value close to 0 indicates that the distribution of the data is approximately symmetrical. Since the value is slightly positive, it suggests a very slight rightward (positive) skew, meaning there might be a few high-value transactions that are pulling the mean to the right, but this effect is minimal.
2. Kurtosis (-1.1435):
A kurtosis of -1.1435 suggests that the export transaction values have lighter tails and a flatter peak compared to a normal distribution. This means there are fewer extreme values (outliers) than we would find in a normal distribution.

Summary : 
The distribution of export transaction values is approximately symmetric with a slight rightward skew, and it is flatter than a normal distribution with fewer extreme values. This might imply that most transactions are clustered around a central value with less variability in the tails.

-> Coefficient of variation (CV) calculated above can provide the following inferences : 
1. High Variability:
A CV above 30% typically indicates high variability relative to the mean. In our case, a CV of 59.67% suggests that the quantity of exported products is quite variable. This means there are significant fluctuations in the quantities being exported.

Summary : 
A CV of 59.67% indicates significant variability in the quantity of exported products, suggesting a need for careful monitoring and potentially adaptive management strategies to address the associated risks and uncertainties.

-> From the results of Levene's test calculated above, we can infer the following:
1. Levene's Test Statistic (1.2582):
A value of approximately 1.26 suggests some degree of variance differences, but this alone doesn't determine significance.
2. P-Value (0.2844):
The p-value indicates the probability of observing the data, or something more extreme, under the null hypothesis (which states that the variances across the groups are equal). A p-value of approximately 0.2844 is higher than the common significance level of 0.05.
3. Fail to Reject the Null Hypothesis:
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. This means there is not enough statistical evidence to conclude that the variances of transaction values differ significantly between the product categories we tested.

Summary : 
The analysis indicates that the variances in transaction values among the different product categories are not significantly different from each other. This suggests that we can assume homogeneity of variances when conducting further statistical analyses (like ANOVA) involving these product categories.

-> From the results of the Shapiro-Wilk test calculated above, we can infer the following:
1. Shapiro-Wilk Test Statistic (0.9598):
The test statistic provides a measure of how closely the sample distribution matches a normal distribution. Values close to 1 indicate normality, while values further from 1 suggest deviations from normality. A statistic of approximately 0.9598 indicates some departure from normality.
2. P-Value (4.0432e-23):
The p-value is extremely small, indicating a very low probability of observing the data (or something more extreme) if the null hypothesis (that the data is normally distributed) is true.
3. Reject the Null Hypothesis:
Since the p-value is much less than the common significance level of 0.05, we reject the null hypothesis. This means there is strong statistical evidence to conclude that the distribution of transaction values is not normally distributed.

Summary : 
The analysis indicates that the distribution of transaction values significantly deviates from normality. This may have implications for further statistical analyses, as many parametric tests (like t-tests or ANOVA) assume normality. 

Managerial Insights and Recommendations:

1. Focus on Key Markets:
Insight: China and the USA represent the largest portion of trade transactions, emphasizing their importance as trade partners.

Recommendation: Strengthen relationships with these key markets by negotiating favorable trade agreements and ensuring that logistics are optimized for these regions. Additionally, explore opportunities to reduce reliance on a few key markets by expanding trade into underrepresented countries.

2. Optimize Shipping Strategies:
Insight: Sea freight is overwhelmingly preferred for larger, bulk shipments, while air freight is used sparingly for high-value goods.

Recommendation: Businesses should continue to use sea freight for bulk shipments to keep costs down. For high-value or time-sensitive goods, consider negotiating better air freight rates or exploring other expedited shipping methods. Incorporating predictive models to determine the most cost-effective shipping methods could further optimize trade logistics.

3. Export Focus for High-Value Goods:
Insight: The analysis shows that exports generally have higher average values compared to imports, which indicates the profitability of certain export products.

Recommendation: Companies should focus on scaling the export of high-value goods by improving production efficiency and targeting markets where these goods are in demand. Additionally, invest in marketing efforts to boost exports in regions where the company has a competitive advantage.

4. Diversification of Trade Partners:
Insight: The dataset shows a concentration of trade in a few countries, with several markets underrepresented.

Recommendation: To mitigate risk and capture untapped opportunities, businesses should diversify trade partners by exploring new markets, particularly in underrepresented regions. Market research and strategic partnerships in these regions could open new revenue streams.

5. Tailored Payment Terms and Shipping Logistics:
Insight: Different shipping methods and payment terms are associated with varying transaction values.

Recommendation: Tailor payment terms and shipping strategies based on transaction size and risk. For high-value transactions, secure payment terms (such as Letters of Credit) and expedited shipping (e.g., air freight) should be prioritized. For lower-value, bulk transactions, businesses can optimize costs by using sea freight and flexible payment options.

6. Data-Driven Decision Making:
Insight: The analysis of trade patterns provides crucial insights into optimizing trade operations.

Recommendation: Leverage advanced data analytics to predict trade trends, optimize shipping routes, and manage inventory. By adopting a data-driven approach, companies can make more informed decisions that enhance profitability and reduce operational risks.

In [1]:
#Loading all the necessary libraries
import os
import pandas as pd
import numpy as np
from scipy.stats import levene
from scipy.stats import skew
from scipy.stats import kurtosis
from scipy.stats import shapiro

In [2]:
#Loading the Data.
df= pd.read_csv('/Users/saumyaraghuvashi/Desktop/Imports_Exports_Dataset.csv')
df

Transaction_ID           Country    Product  \
0      e3e70682-c209-4cac-a29f-6fbed82c07cd          Colombia   describe   
1      f728b4fa-4248-4e3a-8a5d-2f346baa9455             Chile  president   
2      eb1167b3-67a9-4378-bc65-c1e582e2e662            Turkey        far   
3      f7c1bd87-4da5-4709-9471-3d60c8a70639  Christmas Island     agency   
4      e443df78-9558-467f-9ba9-1faf7a024204           Finland     policy   
...                                     ...               ...        ...   
14995  48df15a8-0823-4964-8c16-eddf2756f382  Marshall Islands        not   
14996  31106617-94a6-4646-a001-5e7bd45abc26           Bermuda        air   
14997  ee485839-fbde-4ced-af18-d98f5e863081          Tanzania       show   
14998  5acd54aa-ec8c-4055-be8b-a447861a471c            Tuvalu         TV   
14999  5cc039d0-a052-41fd-bfbb-c9f60c4565ac   North Macedonia       year   

      Import_Export  Quantity    Value        Date     Category  \
0            Export      1979  9506.57  07-12-2023    Machinery   
1            Export      5763  7100.91  04-04-2023     Clothing   
2            Import      5740  2450.34  21-08-2024  Electronics   
3            Export      2592  7226.42  09-05-2022    Furniture   
4            Export      2622  2817.29  03-04-2023  Electronics   
...             ...       ...      ...         ...          ...   
14995        Export      2860  2055.19  09-07-2024    Furniture   
14996        Export      2443  6407.06  18-06-2024    Furniture   
14997        Export      1702  9918.29  30-04-2020         Toys   
14998        Export      8108  9288.57  29-04-2021     Clothing   
14999        Import      5635   561.33  25-12-2019     Clothing   

                     Port  Customs_Code   Weight Shipping_Method  \
0              Robertbury        620537  4248.65             Air   
1                Shahport        927600  4340.81             Air   
2        South Joshuatown        299258  4538.41             Air   
3                Adamfort        996084  4886.14             Air   
4               Juliebury        555981  4406.74             Air   
...                   ...           ...      ...             ...   
14995     South Karenfort        393463  4120.35            Land   
14996         Jeffreyside        484143  1832.71             Air   
14997  North Meganborough        354935  4203.52            Land   
14998           Villafurt        234296  1597.72            Land   
14999           West Gary        766709  2913.82             Air   

                              Supplier        Customer  Invoice_Number  \
0      Garrison, Hubbard and Hendricks       Seth Hall        21000294   
1                            Webb-Mack   Kimberly Ryan        88738432   
2            Mendez, Jones and Johnson      Ryan Silva        89922099   
3                      Schroeder-Smith      Jacob Gray        63216265   
4                        Zimmerman LLC    Amy Stephens        94600248   
...                                ...             ...             ...   
14995                      Smith-Lewis   Darlene Davis        29605073   
14996                      Jones Group      John Ramos        39044695   
14997                    Barnes-Romero  Rebecca Phelps        78492040   
14998       Smith, Allison and Bennett     Scott Yates        20799602   
14999                    Fleming-Miles    Steven Young        79237293   

          Payment_Terms  
0      Cash on Delivery  
1               Prepaid  
2               Prepaid  
3                Net 60  
4      Cash on Delivery  
...                 ...  
14995            Net 60  
14996  Cash on Delivery  
14997            Net 30  
14998  Cash on Delivery  
14999            Net 30  

[15000 rows x 16 columns]

In [3]:
#Displaying the Dimensions of Test Data.
df.shape

(15000, 16)

In [5]:
#Creating and displaying an Unique Sample of 2001 Records using Student Roll Number as Random State.
my_sample=df.sample(n=2001, random_state=55040)
my_sample

Transaction_ID        Country  Product  \
417    21d32e24-bb89-4746-bc3d-a91136a52df9       Tanzania      far   
2687   80746d98-642d-4ecc-a78a-c6464c072c16    Afghanistan  suggest   
4296   a7ef9956-5684-4380-8907-727dc87ea6b5  French Guiana      bed   
12429  30ee19dc-5a86-48d8-b468-5c8ab526c6f5        Senegal    crime   
7696   d165c762-b38a-4c8b-8362-2c3f9c1fb8d2          Congo      now   
...                                     ...            ...      ...   
10633  db6f6def-23fd-43fa-ac46-90eb7dce31a4        Morocco     grow   
6359   243d3f7f-b1bb-47ce-b709-3a44cd5a5c44        Vietnam   street   
2516   9efdfa59-2546-4b21-a4fc-e1fdc600b9b6         Angola     wait   
6138   c4eda72b-b9a9-4872-9cfa-04268c43d0c2         Jordan     role   
13250  6f71d73d-cc44-41cf-82fc-38475147b20c           Mali      put   

      Import_Export  Quantity    Value        Date   Category  \
417          Import      1822  9547.94  05-07-2024   Clothing   
2687         Export      6710  9232.59  11-07-2023  Machinery   
4296         Export      3088  9830.22  22-12-2019  Furniture   
12429        Export      6487  9172.25  04-09-2023  Furniture   
7696         Export      8887  6216.90  22-05-2020   Clothing   
...             ...       ...      ...         ...        ...   
10633        Import      6107  3007.34  13-09-2020   Clothing   
6359         Import      6875  2044.97  14-12-2019  Machinery   
2516         Export      3768  9244.68  27-03-2021  Furniture   
6138         Export      9377  5610.53  15-06-2021       Toys   
13250        Import      6609  2702.35  15-07-2024       Toys   

                   Port  Customs_Code   Weight Shipping_Method  \
417        Barryborough        737864  3080.93             Sea   
2687        Cynthiafort        609756  3040.85             Sea   
4296         Susanmouth        151967    41.40             Air   
12429          New Jose        315564  1809.57            Land   
7696           Lake Jim        579759  3941.10             Sea   
...                 ...           ...      ...             ...   
10633         Lauraland        732453  4259.69             Air   
6359          Greenside        282300  4945.30             Air   
2516          Port Emma        657486  4275.50             Air   
6138     North Johntown        467706  2342.61             Air   
13250  East Theresaview        977516  4341.05            Land   

                       Supplier           Customer  Invoice_Number  \
417       Adams, Beck and Lopez         Joe Hunter        66537138   
2687       Beck, Pena and Boone      Paul Castillo        35347958   
4296             Thompson Group      Andrea Turner         1610967   
12429          Mcdowell-Herrera       Sandra Price        70213981   
7696               Taylor-Tyler        Wendy Ortiz        41523915   
...                         ...                ...             ...   
10633             Lowe and Sons  Dr. Sarah Navarro        95058806   
6359       Silva, Vega and Soto        Amber Hodge        57311937   
2516   Taylor, Cook and Watkins    Amber Hernandez        56443352   
6138               Smith-Parker       Samuel Miles        95555015   
13250  Lewis, Moon and Mckinney      Kevin Bernard        33267576   

          Payment_Terms  
417    Cash on Delivery  
2687             Net 30  
4296            Prepaid  
12429           Prepaid  
7696            Prepaid  
...                 ...  
10633            Net 30  
6359   Cash on Delivery  
2516             Net 30  
6138             Net 30  
13250  Cash on Delivery  

[2001 rows x 16 columns]

In [6]:
#Displaying the Dimensions of Sample Data.
my_sample.shape

(2001, 16)

In [8]:
#Displaying Sample Data Information.
my_sample.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2001 entries, 417 to 13250
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Transaction_ID   2001 non-null   object 
 1   Country          2001 non-null   object 
 2   Product          2001 non-null   object 
 3   Import_Export    2001 non-null   object 
 4   Quantity         2001 non-null   int64  
 5   Value            2001 non-null   float64
 6   Date             2001 non-null   object 
 7   Category         2001 non-null   object 
 8   Port             2001 non-null   object 
 9   Customs_Code     2001 non-null   int64  
 10  Weight           2001 non-null   float64
 11  Shipping_Method  2001 non-null   object 
 12  Supplier         2001 non-null   object 
 13  Customer         2001 non-null   object 
 14  Invoice_Number   2001 non-null   int64  
 15  Payment_Terms    2001 non-null   object 
dtypes: float64(2), int64(3), object(11)
memory usage: 265.8+ 

In [7]:
#Displaying the First 03 Records of the Sample Data.
my_sample.head(3)

Transaction_ID        Country  Product  \
417   21d32e24-bb89-4746-bc3d-a91136a52df9       Tanzania      far   
2687  80746d98-642d-4ecc-a78a-c6464c072c16    Afghanistan  suggest   
4296  a7ef9956-5684-4380-8907-727dc87ea6b5  French Guiana      bed   

     Import_Export  Quantity    Value        Date   Category          Port  \
417         Import      1822  9547.94  05-07-2024   Clothing  Barryborough   
2687        Export      6710  9232.59  11-07-2023  Machinery   Cynthiafort   
4296        Export      3088  9830.22  22-12-2019  Furniture    Susanmouth   

      Customs_Code   Weight Shipping_Method               Supplier  \
417         737864  3080.93             Sea  Adams, Beck and Lopez   
2687        609756  3040.85             Sea   Beck, Pena and Boone   
4296        151967    41.40             Air         Thompson Group   

           Customer  Invoice_Number     Payment_Terms  
417      Joe Hunter        66537138  Cash on Delivery  
2687  Paul Castillo        35347958            Net 30  
4296  Andrea Turner         1610967           Prepaid

In [8]:
#Displaying the List of Names of the Variables.
my_sample.columns

Index(['Transaction_ID', 'Country', 'Product', 'Import_Export', 'Quantity',
       'Value', 'Date', 'Category', 'Port', 'Customs_Code', 'Weight',
       'Shipping_Method', 'Supplier', 'Customer', 'Invoice_Number',
       'Payment_Terms'],
      dtype='object')

In [11]:
#Identifying & Displaying the List of following Variables:
# 1. Index Variables[Transaction_Id]
# 2. Categorical Variables - Nominal Type [Country,Product,Import_Export,Category,Port,Shipping_Method,Supplier,Customer,Payment_Terms]
# 3. Categorical Variables - Ordinal Type.
# 4. Non-Categorical Variables [Quantity,Value,Date,Customs_Code,Weight,Invoice_Number]

In [12]:
#Subset and Display the Non-Categorical Variables.
non_cat=my_sample[['Quantity', 'Value', 'Date', 'Customs_Code', 'Weight','Invoice_Number']]
non_cat

Quantity    Value        Date  Customs_Code   Weight  Invoice_Number
417        1822  9547.94  05-07-2024        737864  3080.93        66537138
2687       6710  9232.59  11-07-2023        609756  3040.85        35347958
4296       3088  9830.22  22-12-2019        151967    41.40         1610967
12429      6487  9172.25  04-09-2023        315564  1809.57        70213981
7696       8887  6216.90  22-05-2020        579759  3941.10        41523915
...         ...      ...         ...           ...      ...             ...
10633      6107  3007.34  13-09-2020        732453  4259.69        95058806
6359       6875  2044.97  14-12-2019        282300  4945.30        57311937
2516       3768  9244.68  27-03-2021        657486  4275.50        56443352
6138       9377  5610.53  15-06-2021        467706  2342.61        95555015
13250      6609  2702.35  15-07-2024        977516  4341.05        33267576

[2001 rows x 6 columns]

In [13]:
#Displaying the Descriptive Statistics of the Non-Categorical Subset.
non_cat.describe()

Quantity        Value   Customs_Code       Weight  Invoice_Number
count  2001.000000  2001.000000    2001.000000  2001.000000    2.001000e+03
mean   4935.177911  5004.899850  553336.820590  2469.226932    5.072527e+07
std    2836.629652  2807.005111  263042.786352  1462.024226    2.912740e+07
min       6.000000   110.750000  101600.000000     2.860000    7.801700e+04
25%    2480.000000  2645.490000  328310.000000  1230.430000    2.590197e+07
50%    4920.000000  4989.120000  546248.000000  2376.450000    5.123842e+07
75%    7356.000000  7415.850000  782526.000000  3755.630000    7.537731e+07
max    9990.000000  9996.810000  999752.000000  4998.730000    9.999782e+07

In [9]:
#Subset and Display the Categorical Variables.
cat_df=my_sample[['Country', 'Product', 'Import_Export', 'Category', 'Port', 'Shipping_Method', 'Supplier','Customer','Payment_Terms']]
cat_df

Country  Product Import_Export   Category              Port  \
417         Tanzania      far        Import   Clothing      Barryborough   
2687     Afghanistan  suggest        Export  Machinery       Cynthiafort   
4296   French Guiana      bed        Export  Furniture        Susanmouth   
12429        Senegal    crime        Export  Furniture          New Jose   
7696           Congo      now        Export   Clothing          Lake Jim   
...              ...      ...           ...        ...               ...   
10633        Morocco     grow        Import   Clothing         Lauraland   
6359         Vietnam   street        Import  Machinery         Greenside   
2516          Angola     wait        Export  Furniture         Port Emma   
6138          Jordan     role        Export       Toys    North Johntown   
13250           Mali      put        Import       Toys  East Theresaview   

      Shipping_Method                  Supplier           Customer  \
417               Sea     Adams, Beck and Lopez         Joe Hunter   
2687              Sea      Beck, Pena and Boone      Paul Castillo   
4296              Air            Thompson Group      Andrea Turner   
12429            Land          Mcdowell-Herrera       Sandra Price   
7696              Sea              Taylor-Tyler        Wendy Ortiz   
...               ...                       ...                ...   
10633             Air             Lowe and Sons  Dr. Sarah Navarro   
6359              Air      Silva, Vega and Soto        Amber Hodge   
2516              Air  Taylor, Cook and Watkins    Amber Hernandez   
6138              Air              Smith-Parker       Samuel Miles   
13250            Land  Lewis, Moon and Mckinney      Kevin Bernard   

          Payment_Terms  
417    Cash on Delivery  
2687             Net 30  
4296            Prepaid  
12429           Prepaid  
7696            Prepaid  
...                 ...  
10633            Net 30  
6359   Cash on Delivery  
2516             Net 30  
6138             Net 30  
13250  Cash on Delivery  

[2001 rows x 9 columns]

In [14]:
#Display the Descriptive Statistics of any 02 Categorical Variables.
country_des_stats = my_sample.groupby('Country').count()
import_export_des_stats = my_sample.groupby('Import_Export').count()
country_des_stats , import_export_des_stats

(                   Transaction_ID  Product  Import_Export  Quantity  Value  \
 Country                                                                      
 Afghanistan                     8        8              8         8      8   
 Albania                         9        9              9         9      9   
 Algeria                        10       10             10        10     10   
 American Samoa                 11       11             11        11     11   
 Andorra                         7        7              7         7      7   
 ...                           ...      ...            ...       ...    ...   
 Wallis and Futuna               6        6              6         6      6   
 Western Sahara                 14       14             14        14     14   
 Yemen                           6        6              6         6      6   
 Zambia                          7        7              7         7      7   
 Zimbabwe                        8        8         

Data Description using Sample Dataset & their Analysis

In [16]:
#Display the Most and Least Preferred Shipping Method.
shipping_method_counts = my_sample['Shipping_Method'].value_counts()

# Most preferred ship mode
most_preferred = shipping_method_counts.idxmax()
most_preferred_count = shipping_method_counts.max()

# Least preferred ship mode
least_preferred = shipping_method_counts.idxmin()
least_preferred_count = shipping_method_counts.min()

# Display the results
print(f"Most Preferred Shipping Method: {most_preferred} with {most_preferred_count} orders.")
print(f"Least Preferred Shipping Method: {least_preferred} with {least_preferred_count} orders.")

Most Preferred Shipping Method: Land with 671 orders.
Least Preferred Shipping Method: Air with 664 orders.


Observation :
We can see that "Land" is the most preferred shipping method with 671 orders, while "Air" is the least preferred with 664 orders. The counts are very close, indicating that there is nearly equal demand for both shipping methods in the dataset.

Managerial Insight : 
1. The near-equal preference for "Land" and "Air" shipping suggests that customers have varied needs regarding delivery speed and cost. 
2. "Land" shipping might be more cost-effective for bulkier items or short distances, whereas "Air" shipping is likely used for time-sensitive or international deliveries.
3. To optimize logistics, management could explore offering more targeted incentives (e.g., discounts on less frequently used methods during low-demand periods) to balance the load between these shipping methods and ensure smoother operations during peak times.

In [15]:
#Country with highest number of transactions 
transactions_bycountry = my_sample.groupby('Country')[['Transaction_ID']].count()
max_transactions = transactions_bycountry.sort_values('Transaction_ID',ascending=False)
max_transactions.head(1)

Transaction_ID
Country                
Korea                21

Observation :
Korea has the highest number of transactions, with 21 recorded.

Managerial Insight : 
Focusing on Korea for future business strategies may yield higher returns, as it represents a significant portion of the company's transaction volume. 
Strengthening relationships and exploring additional opportunities in this market could enhance profitability. 

In [18]:
#Country with least number of transactions 
transactions_bycountry = my_sample.groupby('Country')[['Transaction_ID']].count()
least_transactions = transactions_bycountry.sort_values('Transaction_ID',ascending=True)
least_transactions.head(1)

Transaction_ID
Country                     
Saint Martin               2

Observation : 
Saint Martin has the lower number of transactions, with only 2 recorded suggesting that enhancement solutions are required in this market area.

Managerial Insight : 
Saint Martin, with only 2 transactions, represents an underperforming market compared to other countries. Management should evaluate whether this low activity is due to market potential, lack of resources, or ineffective strategies and decide if targeted marketing or operational adjustments are needed to improve transaction volume in this region.

In [19]:
#Average value of imports vs. exports
imports = my_sample[my_sample['Import_Export'] == 'Import']
exports = my_sample[my_sample['Import_Export'] == 'Export']

average_import_value = imports['Value'].mean()
average_export_value = exports['Value'].mean()

print(f"The average value of imports is {average_import_value:.2f}")
print(f"The average value of exports is {average_export_value:.2f}")

The average value of imports is 5071.74
The average value of exports is 4936.09


Observation: 
The average value of imports ($5071.74) is slightly higher than the average value of exports ($4936.09), indicating a marginal difference between the two trade flows.

Managerial Insight: 
Given the higher average import value, management might focus on optimizing supply chain costs for imports to enhance profitability. Simultaneously, exploring opportunities to boost the average value of exports could enhance overall revenue, balancing the trade dynamics.

In [20]:
#Total value of exports from the top 3 exporting countries and also display the top 3 countries
exports = my_sample[my_sample['Import_Export'] == 'Export']

total_export_values = exports.groupby('Country')['Value'].sum()
top_3_exporting_values = total_export_values.sort_values(ascending=False).head(3)
total_value_top3 = top_3_exporting_values.sum()
print(f"The total value of exports from the top 3 exporting countries is {total_value_top3:.2f}")
print("\nTop 3 Exporting Countries:\n", top_3_exporting_values)

The total value of exports from the top 3 exporting countries is 172289.67

Top 3 Exporting Countries:
 Country
Korea                 69220.69
Russian Federation    52494.50
Cote d'Ivoire         50574.48
Name: Value, dtype: float64


Observation: 
The top 3 exporting countries—Korea, Russian Federation, and Cote d'Ivoire—account for a significant total export value of $172,289.67, with Korea leading the group at $69,220.69.

Managerial Insight: 
Given that a large portion of export value is concentrated in these three countries, management should focus on strengthening trade relationships and logistical efficiencies with Korea, Russia, and Cote d'Ivoire to maximize export growth and streamline operations in these high-performing regions.

In [21]:
#Product having the highest transaction value across all countries
total_value_by_product =my_sample.groupby('Product')['Value'].sum()

product_with_max_value = total_value_by_product.idxmax()
max_value = total_value_by_product.max()

print(f"The product with the highest transaction value is '{product_with_max_value}' with a value of {max_value:.2f}.")

The product with the highest transaction value is 'want' with a value of 53131.60.


Observation: The product 'want' has the highest transaction value across all countries, totaling $53,131.60, indicating strong demand or market preference.

Managerial Insight: Management should leverage this insight by prioritizing marketing and supply chain efforts around the 'want' product. Focusing on enhancing availability, optimizing pricing strategies, and exploring potential partnerships could significantly boost sales and strengthen market positioning.

In [22]:
#Product category with the most number of transactions 

total_value_by_category = my_sample.groupby('Category')['Transaction_ID'].count()
category_with_most_transactions = total_value_by_category.idxmax()
most_transactions = total_value_by_category.max()

print(f"The product category with the most transactions is '{category_with_most_transactions}' with {most_transactions} transactions.")

The product category with the most transactions is 'Machinery' with 432 transactions.


Observation: 
The 'Machinery' category leads with the highest number of transactions at 432, indicating a robust market activity and demand for machinery products.

Managerial Insight: 
Given the high transaction volume in the machinery category, management should consider enhancing inventory management and supply chain efficiencies for these products. Additionally, investing in targeted marketing campaigns and exploring partnerships with key players in this sector could further capitalize on this demand and drive revenue growth.

In [23]:
#Total number of transactions in each country by month

my_sample['Date'] = pd.to_datetime(my_sample['Date'], format='%d-%m-%Y')
my_sample['Year_Month'] = my_sample['Date'].dt.to_period('M')
transactions_by_country_month = my_sample.groupby(['Country', 'Year_Month'])['Transaction_ID'].count()
transactions_by_country_month = transactions_by_country_month.reset_index()
print(transactions_by_country_month)

          Country Year_Month  Transaction_ID
0     Afghanistan    2020-04               1
1     Afghanistan    2020-07               1
2     Afghanistan    2020-08               1
3     Afghanistan    2021-01               1
4     Afghanistan    2021-03               1
...           ...        ...             ...
1876     Zimbabwe    2021-12               1
1877     Zimbabwe    2022-06               1
1878     Zimbabwe    2022-10               1
1879     Zimbabwe    2023-03               1
1880     Zimbabwe    2024-01               1

[1881 rows x 3 columns]


Observation: 
The sample dataset reveals a total of 1,881 transactions categorized by country and month, indicating diverse trading activity across multiple countries over several months.

Managerial Insight:
Management should analyze transaction trends over time to identify peak trading months and regional patterns. This information can inform inventory planning, resource allocation, and targeted marketing strategies, allowing the company to optimize operations and enhance customer satisfaction during high-demand periods.

In [24]:
#Display the country having the highest average value per transaction
average_value_per_transaction = my_sample.groupby('Country')['Value'].mean()

sorted_average_value = average_value_per_transaction.sort_values(ascending=False).head(1)

print("Country with the highest average value per transaction:\n")
print(sorted_average_value)

Country with the highest average value per transaction:

Country
Saint Martin    8689.735
Name: Value, dtype: float64


Observation: 
Saint Martin has the highest average value per transaction at $8,689.74, indicating that transactions from this country tend to involve higher-value goods or services.

Managerial Insight: 
Management should consider focusing on Saint Martin as a key market for high-value offerings. Strengthening relationships with local partners and tailoring marketing strategies to emphasize premium products could enhance sales and profitability in this lucrative market segment.

In [26]:
#Product category with the least number of transactions 

total_value_by_category = my_sample.groupby('Category')['Transaction_ID'].count()
category_with_least_transactions = total_value_by_category.idxmin()
least_transactions = total_value_by_category.min()

print(f"The product category with the most transactions is '{category_with_least_transactions}' with {least_transactions} transactions.")

The product category with the most transactions is 'Electronics' with 373 transactions.


Observation: 
The 'Electronics' category has the least number of transactions, totaling 373, suggesting lower market activity or demand compared to other categories.

Managerial Insight: 
Management should investigate the reasons behind the low transaction volume in the electronics category. This may involve conducting market research to identify potential barriers to sales, enhancing product offerings, or refining marketing strategies to stimulate demand and improve transaction levels in this segment.

In [29]:
#Display the Most preferred payment term method

payment_term_counts = my_sample['Payment_Terms'].value_counts()

most_preferred_payment_term = payment_term_counts.idxmax()
most_preferred_payment_term_count = payment_term_counts.max()


print(f"Most preferred payment terms method: {most_preferred_payment_term} with {most_preferred_payment_term_count} orders.")

Most preferred payment terms method: Net 60 with 528 orders.


Observation: 
The most preferred payment term method is "Net 60," with a total of 528 orders, indicating a strong preference among clients for this extended payment option.

Managerial Insight: 
Management should consider standardizing "Net 60" as the default payment term to enhance customer satisfaction and potentially increase sales. Additionally, understanding the reasons behind this preference can help tailor financial offerings and improve client relationships, further optimizing cash flow management.

In [30]:
#Display the least preferred payment term method

payment_term_counts = my_sample['Payment_Terms'].value_counts()

least_preferred_payment_term = payment_term_counts.idxmin()
least_preferred_payment_term_count = payment_term_counts.min()


print(f"Least preferred payment terms method: {least_preferred_payment_term} with {least_preferred_payment_term_count} orders.")

Least preferred payment terms method: Cash on Delivery with 489 orders.


Observation: 
The least preferred payment term method is "Cash on Delivery," with only 489 orders, indicating a lack of confidence or convenience among customers for this payment option.

Managerial Insight: 
Management should analyze the factors contributing to the low popularity of "Cash on Delivery" and consider providing additional payment options or incentives to encourage its use. Improving the perception of security and reliability for this payment method could help increase its adoption and enhance customer satisfaction.

In [45]:
#Display the product category with the highest total weight being shipped 

total_value_by_category = my_sample.groupby('Category')['Weight'].sum()
category_with_max_weight = total_value_by_category.idxmax()
max_weight = round(total_value_by_category.max(),2)

print(f"The product category with the highest total weight being shipped is '{category_with_max_weight}' with a total weight of {max_weight} units.")

The product category with the highest total weight being shipped is 'Machinery' with a total weight of 1122035.03 units.


Observation: 
The 'Machinery' category has the highest total weight being shipped, totaling 1,122,035.03 units, indicating significant volume in this product line.

Managerial Insight: 
Management should optimize logistics and supply chain management specifically for the machinery category to ensure efficient handling and transportation of these heavy products. Additionally, focusing on inventory management and potentially exploring bulk shipping options could reduce costs and enhance service delivery for this high-volume category.

In [47]:
#Display the product category with the least total weight being shipped 

total_value_by_category = my_sample.groupby('Category')['Weight'].sum()
category_with_least_weight = total_value_by_category.idxmin()
least_weight = round(total_value_by_category.min(),2)

print(f"The product category with the least total weight being shipped is '{category_with_least_weight}' with a total weight of {least_weight} units.")

The product category with the least total weight being shipped is 'Electronics' with a total weight of 887055.55 units.


Observation: 
The 'Electronics' category has the least total weight being shipped, amounting to 887,055.55 units, indicating a relatively lighter product profile compared to other categories.

Managerial Insight: 
Management should evaluate the reasons behind the low shipping weight in the electronics category, which may include product design or market preferences. Focusing on optimizing packaging and shipping processes for lighter items could reduce shipping costs and enhance competitiveness in pricing, potentially increasing sales in this category.

In [48]:
#Display the country having the least average value per transaction

average_value_per_transaction = my_sample.groupby('Country')['Value'].mean()

least_average_value = average_value_per_transaction.sort_values(ascending=True).head(1)

print("Country with the least average value per transaction:\n")
print(least_average_value)

Country with the least average value per transaction:

Country
Botswana    2186.042
Name: Value, dtype: float64


Observation: 
Botswana has the least average value per transaction at $2,186.04, indicating that the transactions from this country involve relatively low-value goods or services.

Managerial Insight: 
Management should investigate the factors contributing to the low average transaction value in Botswana. By understanding market dynamics and customer preferences, the company can explore opportunities to introduce higher-value products or services, thereby enhancing revenue and improving the overall transaction value in this market.

In [49]:
#Total value of imports from the top 3 importing countries and displaying the top 3 countries
imports = my_sample[my_sample['Import_Export'] == 'Import']

total_import_values = imports.groupby('Country')['Value'].sum()
top_3_importing_values = total_import_values.sort_values(ascending=False).head(3)
total_value_top3 = top_3_importing_values.sum()
print(f"The total value of imports from the top 3 importing countries is {total_value_top3:.2f}")
print("\nTop 3 Importing Countries:\n", top_3_importing_values)

The total value of imports from the top 3 importing countries is 215433.92

Top 3 Importing Countries:
 Country
Congo       91274.00
Tanzania    62821.17
Pakistan    61338.75
Name: Value, dtype: float64


Observation: 
The total value of imports from the top three importing countries—Congo, Tanzania, and Pakistan—amounts to $215,433.92, with Congo being the largest contributor at $91,274.00.

Managerial Insight: 
Management should focus on strengthening relationships and logistics with these key importing countries, particularly Congo, to ensure a steady supply chain. By optimizing import strategies and potentially negotiating better terms, the company can enhance its operational efficiency and capitalize on the significant import value from these regions.

In [50]:
#Product having the least transaction value across all countries
total_value_by_product = my_sample.groupby('Product')['Value'].sum()

product_with_least_value = total_value_by_product.idxmin()
least_value = total_value_by_product.min()

print(f"The product with the least transaction value is '{product_with_least_value}' with a value of {least_value:.2f}.")

The product with the least transaction value is 'whole' with a value of 112.88.


Observation: 
The product with the least transaction value is 'whole', totaling only $112.88, indicating it is likely underperforming in the market.

Managerial Insight: 
Management should assess the reasons behind the low transaction value of 'whole.' This may involve evaluating market demand, pricing strategies, or marketing efforts. By identifying and addressing the factors contributing to its poor performance, the company could enhance its sales strategy, potentially leading to improved revenue from this product category.

In [54]:
#Supplier who has conducted the highest number of transactions

transaction_count_by_supplier = my_sample.groupby('Supplier')['Transaction_ID'].count()

top_supplier = transaction_count_by_supplier.idxmax()
max_transactions = transaction_count_by_supplier.max()

print(f"The supplier with the highest number of transactions is '{top_supplier}' with {max_transactions} transactions.")

The supplier with the highest number of transactions is 'Brown PLC' with 5 transactions.


Observation: 
The supplier with the highest number of transactions is 'Brown PLC,' which has conducted 5 transactions, indicating a potential strong partnership or reliance on this supplier for operations.

Managerial Insight: 
Management should consider evaluating the terms and efficiency of its relationship with Brown PLC. Given their significant involvement, optimizing this partnership could enhance supply chain efficiency, improve pricing, and potentially lead to better product offerings, thereby strengthening overall business performance.

In [80]:
#Transactions made in each port
transactions_per_port = my_sample['Port'].value_counts()
transactions_per_port

Lake Jessica        5
Port Amanda         5
Port David          4
West Matthew        4
Port Jennifer       4
                   ..
Sheilaton           1
West Kevinview      1
Kirkchester         1
East Christopher    1
Andersonton         1
Name: Port, Length: 1842, dtype: int64

Observation: 
The dataset reveals that Lake Jessica and Port Amanda each account for the highest number of transactions (5 each), while many ports, including Sheilaton and West Kevinview, have only a single transaction, indicating uneven transaction distribution across ports.

Managerial Insight: 
Management should analyze the efficiency and throughput of the most frequently used ports, like Lake Jessica and Port Amanda, to optimize logistics and shipping processes. Additionally, assessing the underutilized ports could unveil opportunities for diversification in shipping routes, potentially reducing congestion and improving overall operational resilience.

In [61]:
#The total weight of exported products versus imported products

imports = my_sample[my_sample['Import_Export'] == 'Import']
exports = my_sample[my_sample['Import_Export'] == 'Export']

total_imported_weight = imports['Weight'].sum()
total_exported_weight = exports['Weight'].sum()

print(f"The total weight of exported products is {total_exported_weight:.2f} and total weight of imported products is {total_imported_weight:.2f}.")

The total weight of exported products is 2405510.86 and total weight of imported products is 2535412.23.


Observation: 
The total weight of imported products (2,535,412.23 units) exceeds that of exported products (2,405,510.86 units), indicating a higher reliance on imports compared to exports.

Managerial Insight: 
This imbalance suggests a need for management to evaluate its supply chain strategy, focusing on enhancing local production capabilities or diversifying export markets. By reducing dependency on imports, the company could improve profitability and strengthen its market position against potential supply chain disruptions.

In [63]:
#The average quantity of imports and exports per country

exports = my_sample[my_sample['Import_Export'] == 'Export']
imports = my_sample[my_sample['Import_Export'] == 'Import']

average_export_quantity_by_country = exports.groupby('Country')['Quantity'].mean()

average_import_quantity_by_country = imports.groupby('Country')['Quantity'].mean()

average_export_quantity_by_country = average_export_quantity_by_country.reset_index()
average_import_quantity_by_country = average_import_quantity_by_country.reset_index()

print("Average Export Quantity per Country:\n", average_export_quantity_by_country)
print("\nAverage Import Quantity per Country:\n", average_import_quantity_by_country)

Average Export Quantity per Country:
                Country  Quantity
0          Afghanistan   8148.50
1              Albania   6164.75
2              Algeria   6054.80
3       American Samoa   2015.50
4              Andorra   8365.00
..                 ...       ...
237  Wallis and Futuna   3797.00
238     Western Sahara   4456.75
239              Yemen   3942.60
240             Zambia   5998.00
241           Zimbabwe   4683.00

[242 rows x 2 columns]

Average Import Quantity per Country:
                Country     Quantity
0          Afghanistan  4296.166667
1              Albania  4671.000000
2              Algeria  6095.200000
3       American Samoa  4367.714286
4              Andorra  5478.200000
..                 ...          ...
235  Wallis and Futuna  7809.500000
236     Western Sahara  5355.900000
237              Yemen  2449.000000
238             Zambia  6147.666667
239           Zimbabwe  6341.600000

[240 rows x 2 columns]


Observation: 
The average export quantity per country varies significantly, with countries like Afghanistan averaging 8,148.50 units exported, while others, such as American Samoa, export an average of only 2,015.50 units. In contrast, the average import quantity also shows variability, with Afghanistan averaging 4,296.17 units imported.

Managerial Insight: 
This data highlights the need for targeted market strategies. Countries with higher export averages could be potential growth markets , while those with lower import averages may require a deeper understanding of local demand dynamics. Adjusting inventory and supply chain strategies to align with these insights could enhance operational efficiency and improve market penetration.

In [64]:
#Customer having the highest transaction value 

total_transaction_value_by_customer = my_sample.groupby('Customer')['Value'].sum()

top_customer = total_transaction_value_by_customer.idxmax()
highest_transaction_value = total_transaction_value_by_customer.max()


print(f"The customer with the highest transaction value is '{top_customer}' with a total transaction value of {highest_transaction_value} units.")

The customer with the highest transaction value is 'Michael Miller' with a total transaction value of 38719.84 units.


Observation: 
Michael Miller stands out as the customer with the highest transaction value, totaling 38,719.84 units. This indicates a strong business relationship, possibly suggesting that he is a key client for the company.

Managerial Insight: 
Given Michael Miller's significant contribution to transaction value, it would be prudent to foster this relationship further through personalized communication, loyalty incentives, or tailored offerings. Engaging with top customers can enhance customer satisfaction and retention, ultimately driving revenue growth for the business.

In [68]:
#The most frequent customs code used in transactions

customs_code_count = my_sample['Customs_Code'].value_counts()

most_used_customs_code = customs_code_count.idxmax()
most_used_customs_code_count = customs_code_count.max()


print(f"The customs code used most frequently is {most_used_customs_code} with {most_used_customs_code_count} occurrences.")

The customs code used most frequently is 875434 with 2 occurrences.


Observation: 
The customs code 875434 is the most frequently used in transactions, occurring twice. This may indicate a common product or service type associated with this code.

Managerial Insight: 
Given the frequent use of customs code 875434, management should analyze the associated products or services to identify opportunities for optimization, such as bulk purchasing agreements or targeted marketing strategies. Additionally, understanding the reasons behind its prevalence can help streamline processes and enhance compliance with customs regulations.

In [69]:
#The total value of transactions grouped by payment terms

total_value_by_payment_terms = my_sample.groupby('Payment_Terms')['Value'].sum()

total_value_by_payment_terms = total_value_by_payment_terms.reset_index()

print(total_value_by_payment_terms)

      Payment_Terms       Value
0  Cash on Delivery  2451569.15
1            Net 30  2445624.00
2            Net 60  2608667.64
3           Prepaid  2508943.81


Observation: 
The total value of transactions shows that "Net 60" payment terms have the highest transaction value at approximately 2,608,667.64 units, closely followed by "Prepaid" at around 2,508,943.81 units. "Cash on Delivery" and "Net 30" have lower transaction values.

Managerial Insight: 
The dominance of "Net 60" payment terms suggests that customers may prefer longer payment cycles, which could affect cash flow. Management should evaluate the implications of these terms on liquidity and consider strategies to encourage the adoption of more immediate payment methods, like "Prepaid," to enhance cash flow and reduce financial risk.

In [70]:
#How does the value of transactions vary by product category?

transaction_value_by_category = my_sample.groupby('Category')['Value'].agg(['sum', 'mean', 'count']).reset_index()

transaction_value_by_category.columns = ['Category', 'Total Transaction Value', 'Average Transaction Value', 'Transaction Count']

print(transaction_value_by_category)

      Category  Total Transaction Value  Average Transaction Value  \
0     Clothing               1895982.57                4873.991183   
1  Electronics               1947800.81                5221.986086   
2    Furniture               2044712.35                5111.780875   
3    Machinery               2115229.61                4896.364838   
4         Toys               2011079.26                4941.226683   

   Transaction Count  
0                389  
1                373  
2                400  
3                432  
4                407  


Observation: 
The analysis reveals that the "Machinery" category has the highest total transaction value at approximately 2,115,229.61 units, despite having a slightly lower average transaction value compared to categories like "Electronics" and "Furniture." All categories show a significant number of transactions, with "Clothing" having the highest transaction count.

Managerial Insight: 
The prominence of the "Machinery" category in total transaction value indicates a strong market demand, suggesting that resources should be allocated to enhance supply chain efficiency and product availability in this category. Additionally, the relatively high average transaction values in categories like "Electronics" may present opportunities for targeted marketing strategies to boost sales and customer engagement further.

In [74]:
#How has the transaction count for imports and exports changed over time (monthly/yearly)?

my_sample['Date'] = pd.to_datetime(my_sample['Date'])

my_sample['Year'] = my_sample['Date'].dt.year
my_sample['Month'] = my_sample['Date'].dt.month

yearly_transaction_count = my_sample.groupby(['Year', 'Import_Export'])['Transaction_ID'].count().reset_index()

monthly_transaction_count = my_sample.groupby(['Year', 'Month', 'Import_Export'])['Transaction_ID'].count().reset_index()

print("Yearly Transaction Count (Imports vs Exports):\n", yearly_transaction_count)
print("\nMonthly Transaction Count (Imports vs Exports):\n", monthly_transaction_count)

Yearly Transaction Count (Imports vs Exports):
     Year Import_Export  Transaction_ID
0   2019        Export              75
1   2019        Import              65
2   2020        Export             200
3   2020        Import             211
4   2021        Export             174
5   2021        Import             192
6   2022        Export             214
7   2022        Import             199
8   2023        Export             198
9   2023        Import             216
10  2024        Export             125
11  2024        Import             132

Monthly Transaction Count (Imports vs Exports):
      Year  Month Import_Export  Transaction_ID
0    2019      9        Export              20
1    2019      9        Import              14
2    2019     10        Export              20
3    2019     10        Import              12
4    2019     11        Export              19
..    ...    ...           ...             ...
117  2024      7        Import              24
118  2024      8   

Observation: 
The yearly transaction count indicates a general upward trend in both imports and exports from 2019 to 2024, with imports consistently outpacing exports. Notably, in 2022, both imports and exports saw a peak in transaction counts, while exports dipped in 2024.

Managerial Insight: 
The growth in import transactions suggests a robust demand for imported goods, indicating potential opportunities to strengthen supplier relationships and optimize inventory management. Additionally, the decline in export transactions in 2024 could signal the need for a strategic review of international market engagement and targeted marketing efforts to boost export performance.

In [79]:
#Most commonly exported product

exports = my_sample[my_sample['Import_Export'] == 'Export']

export_product_count = exports['Product'].value_counts()

most_exported_product = export_product_count.idxmax()
most_exported_product_count = export_product_count.max()


print(f"The most commonly exported product is '{most_exported_product}' with {most_exported_product_count} exports.")


The most commonly exported product is 'above' with 5 exports.


Observation: 
The data reveals that the product labeled 'above' is the most commonly exported item, with a total of 5 exports. This indicates a concentrated preference for this product in international markets, albeit with a relatively low export volume.

Managerial Insight: 
The limited number of exports for the most commonly exported product suggests an opportunity to expand marketing efforts and enhance distribution channels for 'above.' By analyzing market demand and potential barriers to entry in new regions, management could leverage this product to boost overall export volume and revenue.

In [81]:
#The mean, median, and mode of transaction values across all transactions

mean_value = my_sample['Value'].mean()
median_value = my_sample['Value'].median()
mode_value = my_sample['Value'].mode()[0]

print("Mean:", mean_value)
print("Median:", median_value)
print("Mode:", mode_value)

Mean: 5004.899850074956
Median: 4989.12
Mode: 3673.92


Observation: 
The mean transaction value is approximately 5004.90, while the median is 4989.12, and the mode is 3673.92. This suggests that most transactions are centered around similar values, with a slight skew toward lower transaction amounts.

Managerial Insight: 
With the mode significantly lower than the mean, it indicates that a large number of transactions are at the lower end of the value spectrum. Management might focus on increasing the transaction value for the majority of lower-end transactions by upselling or offering value-added services to boost overall revenue.

In [82]:
#the mean and median transaction value by product category

mean_median_by_category = my_sample.groupby('Category')['Value'].agg(['mean', 'median'])
print(mean_median_by_category)

                    mean    median
Category                          
Clothing     4873.991183  4976.500
Electronics  5221.986086  5060.600
Furniture    5111.780875  5167.390
Machinery    4896.364838  4875.265
Toys         4941.226683  4909.810


Observation:
The product categories show relatively similar mean and median transaction values, with Electronics having the highest mean value (5221.99) and Furniture having the highest median value (5167.39).

Managerial Insight:
Electronics, with the highest mean transaction value, may offer an opportunity to focus on high-value transactions. Management could explore targeted marketing strategies to increase sales in this category, as it has the potential to contribute the most revenue per transaction.

In [84]:
#The range, standard deviation, and variance of the weight of shipped products

weight_range = my_sample['Weight'].max() - my_sample['Weight'].min()
weight_std = my_sample['Weight'].std()
weight_var = my_sample['Weight'].var()

print("Range of Weight:", weight_range)
print("Standard Deviation of Weight:", weight_std)
print("Variance of Weight:", weight_var)

Range of Weight: 4995.87
Standard Deviation of Weight: 1462.0242255032833
Variance of Weight: 2137514.8359584752


Observation: 
The weight of shipped products shows significant variability, with a large range (4995.87 units), a high standard deviation (1462.02), and substantial variance (2137514.84), indicating wide fluctuations in shipment sizes.

Managerial Insight: 
Given the high variability in shipment weights, optimizing logistics based on product weight could reduce shipping costs. Management might consider implementing weight-based pricing models or streamlining shipments to achieve more consistent load sizes.

In [107]:
#the skewness and kurtosis of transaction values for exports

export_data = my_sample[my_sample['Import_Export'] == 'Export']
skewness = skew(export_data['Value'])
kurt = kurtosis(export_data['Value'])

print("Skewness of Export Transaction Values:", skewness)
print("Kurtosis of Export Transaction Values:", kurt)

Skewness of Export Transaction Values: 0.02780808295617856
Kurtosis of Export Transaction Values: -1.1435359285374704


Observation: 
The skewness of export transaction values is close to zero (0.0278), suggesting a nearly symmetrical distribution. The negative kurtosis (-1.1435) indicates that the distribution is flatter than a normal distribution, with fewer extreme values.

Managerial Insight: 
Since export transaction values are relatively symmetrically distributed and lack extreme outliers, management can expect consistent transaction values. This stability could aid in more reliable revenue forecasting and budgeting for export-related operations.

In [104]:
#the coefficient of variation for the quantity of exported products

export_data = my_sample[my_sample['Import_Export'] == 'Export']
mean = np.mean(export_data['Quantity'])
std_dev = np.std(export_data['Quantity'], ddof=1) 
coefficient_of_variation = (std_dev / mean) * 100
print ("Coefficient of variation:", coefficient_of_variation)

Coefficient of variation: 59.67089373823173


Observation: 
The coefficient of variation for the quantity of exported products is 59.67%, indicating a moderate to high level of variability in the quantities of products exported.

Managerial Insight:
The high variability in export quantities suggests that export volumes are inconsistent across transactions. This could signal the need for better demand forecasting or supply chain optimization to reduce fluctuations and maintain steadier export flows.

In [108]:
#Levene’s test to determine if the variance in transaction values differs significantly between product categories

grouped_data = [group['Value'].values for name, group in my_sample.groupby('Category')]
statistic, p_value = levene(*grouped_data)
print(f"Levene's Test Statistic: {statistic}")
print(f"P-Value: {p_value}")

alpha = 0.05  # significance level
if p_value < alpha:
    print("Reject the null hypothesis: Variances are significantly different.")
else:
    print("Fail to reject the null hypothesis: Variances are not significantly different.")

Levene's Test Statistic: 1.2581638894117086
P-Value: 0.28435124949206964
Fail to reject the null hypothesis: Variances are not significantly different.


Observation:
Levene's test reveals a p-value of 0.284, indicating that the variances in transaction values between product categories are not significantly different at a 5% significance level.

Managerial Insight: 
Since the transaction value variances across product categories are relatively consistent, pricing or cost management strategies can be uniformly applied across categories without concern for drastic variability.

In [110]:
#Shapiro-Wilk test to check if the distribution of transaction values is normally distributed

statistic, p_value = shapiro(my_sample['Value'])

print(f"Shapiro-Wilk Test Statistic: {statistic}")
print(f"P-Value: {p_value}")

alpha = 0.05  # significance level
if p_value < alpha:
    print("Reject the null hypothesis: The distribution is not normally distributed.")
else:
    print("Fail to reject the null hypothesis: The distribution is normally distributed.")

Shapiro-Wilk Test Statistic: 0.9597718715667725
P-Value: 4.0431620503924595e-23
Reject the null hypothesis: The distribution is not normally distributed.


Observation: 
The Shapiro-Wilk test shows a p-value of approximately 0, indicating that the transaction values are not normally distributed.

Managerial Insight:
Since the distribution of transaction values is not normal, it may be more effective to use non-parametric statistical methods for analysis and decision-making, particularly when assessing transaction trends and customer behaviors.

# Analysis : 

-> The mean, median, and mode of transaction values across all transactions calculated above provides valuable insights into the distribution of transaction values. Here’s what we can infer from the provided values:
1. Mean (5004.90):
The mean is the average of all transaction values. A mean of approximately 5004.90 suggests that the average transaction value is relatively high. However, the mean can be influenced by outliers or skewed data, so it’s essential to consider it alongside the median and mode.
2. Median (4989.12):
The median is the middle value when all transaction values are ordered. A median of 4989.12 indicates that half of the transactions are below this value and half are above. Since the median is very close to the mean, this suggests a relatively symmetrical distribution of transaction values, but it does not rule out the presence of outliers.
3. Mode (3673.92):
The mode is the most frequently occurring transaction value in the dataset. A mode of 3673.92 indicates that this value appears more often than any other. The mode being lower than both the mean and the median suggests that there may be a significant number of lower-value transactions, which could be influencing the average upward.

Overall Inferences:
Distribution Shape: The fact that the mean is higher than the median suggests a slight right skew in the distribution (more higher-value transactions). However, the closeness of the mean and median implies that the skew might not be extreme.
Presence of Outliers: The relatively high mean compared to the mode suggests that there are higher transaction values (possibly outliers) that are affecting the average. This could warrant further investigation to understand these high-value transactions better.
Transaction Value Trends: The mode indicates that a significant number of transactions are clustered around the value of 3673.92, which could be relevant for pricing strategies or market analysis.


-> From the mean and median transaction values calculated by product category, we can draw several inferences:
1. Clothing:
Mean (4873.99) vs. Median (4976.50):
The mean is slightly lower than the median, suggesting a slight leftward (negative) skew in the distribution. This may indicate that there are some lower-value transactions that are pulling the average down.
2. Electronics:
Mean (5221.99) vs. Median (5060.60):
Here, the mean is higher than the median, indicating a slight rightward (positive) skew. This suggests that there are higher-value transactions in electronics that are driving the average up.
3. Furniture:
Mean (5111.78) vs. Median (5167.39):
Similar to electronics, the mean is lower than the median. This suggests that there might be lower-value transactions affecting the average, leading to a slight leftward skew.
4. Machinery:
Mean (4896.36) vs. Median (4875.27):
The mean is slightly higher than the median, indicating a very mild rightward skew, but it’s close enough to suggest a relatively symmetrical distribution around the center.
5. Toys:
Mean (4941.23) vs. Median (4909.81):
The mean is again slightly higher than the median, suggesting a mild rightward skew similar to machinery.

Overall Inferences:
Skewness Insights:
Categories such as clothing and furniture show a slight leftward skew, indicating some low-value outliers. In contrast, electronics, machinery, and toys display a slight rightward skew, suggesting higher-value outliers may be present.

Central Tendency Comparison:
The mean and median values provide insights into the central tendency of each category. For example, electronics has the highest mean and median, indicating that this category tends to have the highest transaction values overall, followed by furniture and toys.

Data Distribution Considerations:
The differences between mean and median in each category suggest that when analyzing transaction values, it might be helpful to consider both measures of central tendency to understand the impact of skewed distributions and potential outliers.

-> From the range, standard deviation, and variance of the weights of shipped products calculated above, we can infer the following:
1. Range (4995.87):
A range of 4995.87 suggests a large spread between the lightest and heaviest products. This indicates a diverse set of product weights, which could have implications for shipping costs, handling, and storage requirements.
2. Standard Deviation (1462.02):
A standard deviation of 1462.02 indicates a high degree of variability in the weights of the products. This suggests that weights are widely dispersed around the mean, meaning some products are significantly lighter or heavier than the average.
3. Variance (2137514.84):
A variance of 2137514.84 confirms the high variability in product weights. 

Overall Inferences:
High Variability: The large range, along with the high standard deviation and variance, indicates that there is considerable variability in the weights of shipped products. This could mean that your product line includes both very light and very heavy items.
Impact on Shipping: The variability in weight might affect shipping strategies, costs, and logistics. For example, heavier products may require different handling procedures or shipping methods compared to lighter ones.
Potential for Optimization: If certain weights are significantly more common, it might be worth exploring whether shipping practices can be optimized for those weights, such as bulk shipping or specialized packaging.

Summary : 
Overall, the metrics suggest a diverse and variable distribution of product weights, which could have significant implications for logistics, costs, and operational efficiency. The high degree of variability indicates a need for careful planning and consideration in shipping and handling practices.

-> The skewness and kurtosis values calculated above provide insights into the distribution of export transaction values. Here’s what we can infer from each:
1. Skewness (0.0278):
A skewness value close to 0 indicates that the distribution of the data is approximately symmetrical. Since the value is slightly positive, it suggests a very slight rightward (positive) skew, meaning there might be a few high-value transactions that are pulling the mean to the right, but this effect is minimal.
2. Kurtosis (-1.1435):
A kurtosis of -1.1435 suggests that the export transaction values have lighter tails and a flatter peak compared to a normal distribution. This means there are fewer extreme values (outliers) than we would find in a normal distribution.

Summary : 
The distribution of export transaction values is approximately symmetric with a slight rightward skew, and it is flatter than a normal distribution with fewer extreme values. This might imply that most transactions are clustered around a central value with less variability in the tails.

-> Coefficient of variation (CV) calculated above can provide the following inferences : 
1. High Variability:
A CV above 30% typically indicates high variability relative to the mean. In our case, a CV of 59.67% suggests that the quantity of exported products is quite variable. This means there are significant fluctuations in the quantities being exported.

Summary : 
A CV of 59.67% indicates significant variability in the quantity of exported products, suggesting a need for careful monitoring and potentially adaptive management strategies to address the associated risks and uncertainties.

-> From the results of Levene's test calculated above, we can infer the following:
1. Levene's Test Statistic (1.2582):
A value of approximately 1.26 suggests some degree of variance differences, but this alone doesn't determine significance.
2. P-Value (0.2844):
The p-value indicates the probability of observing the data, or something more extreme, under the null hypothesis (which states that the variances across the groups are equal). A p-value of approximately 0.2844 is higher than the common significance level of 0.05.
3. Fail to Reject the Null Hypothesis:
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. This means there is not enough statistical evidence to conclude that the variances of transaction values differ significantly between the product categories we tested.

Summary : 
The analysis indicates that the variances in transaction values among the different product categories are not significantly different from each other. This suggests that we can assume homogeneity of variances when conducting further statistical analyses (like ANOVA) involving these product categories.

-> From the results of the Shapiro-Wilk test calculated above, we can infer the following:
1. Shapiro-Wilk Test Statistic (0.9598):
The test statistic provides a measure of how closely the sample distribution matches a normal distribution. Values close to 1 indicate normality, while values further from 1 suggest deviations from normality. A statistic of approximately 0.9598 indicates some departure from normality.
2. P-Value (4.0432e-23):
The p-value is extremely small, indicating a very low probability of observing the data (or something more extreme) if the null hypothesis (that the data is normally distributed) is true.
3. Reject the Null Hypothesis:
Since the p-value is much less than the common significance level of 0.05, we reject the null hypothesis. This means there is strong statistical evidence to conclude that the distribution of transaction values is not normally distributed.

Summary : 
The analysis indicates that the distribution of transaction values significantly deviates from normality. This may have implications for further statistical analyses, as many parametric tests (like t-tests or ANOVA) assume normality. 